lib/braze_api/endpoints/subscription_groups/statuses.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module BrazeAPI
+  module Endpoints
+    module SubscriptionGroups
+      # Methods to call the subscription/status endpoint from a client instance
+      module Statuses
+        PATH = '/subscription/user/status'
+        # The method to get a users subscriptions status
+        # Args object will look like:
+        # {subscription_state: 'subscribed', subscription_group_id: id, external_id: uuid}
+        # with either an external_id or an email and the subscription state
```

Comment:
From my understanding of [the docs](https://www.braze.com/docs/api/endpoints/subscription_groups/get_list_user_subscription_groups/), this should be:
```suggestion
      # Methods to call the subscription/status endpoint from a client instance
      module Statuses
        PATH = '/subscription/user/status'
        # The method to get a users subscriptions status
        # Args object will look like:
        # {external_id: uuid, limit: integer, offset: integer}
        # with either an external_id or an email and the subscription state
```

Reply:
Hi @Robbie-Glynn thank you so much for reviewing this

The limit and offset properties can be both be used that's right, (but we also need to pass the subscription_group_id) in the `/subscription/user/status` endpoint. The endpoint will return the status for an specific given `subscription_group_id` for a given `user`.

I like the idea of restructuring the folders I just didn't wanted to change existing code created by other authors and being used already by developers but happy to do that directory structure if you think is good idea.